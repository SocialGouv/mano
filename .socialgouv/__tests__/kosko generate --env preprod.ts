//

import { getEnvManifests } from "@socialgouv/kosko-charts/testing";
import { project } from "@socialgouv/kosko-charts/testing/fake/gitlab-ci.env";

jest.setTimeout(1000 * 60);
test("kosko generate --preprod", async () => {
  process.env.HARBOR_PROJECT = "mano";
  expect(
    await getEnvManifests("preprod", "", {
      ...project("mano").preprod,
      KUBE_NAMESPACE: "mano-186-preprod-dev2",
      // RANCHER_PROJECT_ID: "c-bd7z2:p-dtlsm",
    })
  ).toMatchSnapshot();
});
