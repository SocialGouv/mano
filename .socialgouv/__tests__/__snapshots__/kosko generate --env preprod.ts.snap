// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`kosko generate --preprod 1`] = `
"---
apiVersion: v1
kind: Namespace
metadata:
  annotations:
    socialgouv/creator: autodevops
    field.cattle.io/creatorId: gitlab
    field.cattle.io/projectId: ''
    git/branch: refs/tags/v1.2.3
    git/remote: socialgouv/mano
    app.github.com/job: '5678'
    app.github.com/ref: refs/tags/v1.2.3
    app.github.com/repo: socialgouv/mano
    app.github.com/run: '1234'
    app.github.com/sha: '0123456'
  labels:
    azure-pg-admin-user: mano
    application: mano
    component: mano
    owner: mano
    team: mano
    cert: wildcard
  name: mano-preprod
---
apiVersion: apps/v1
kind: Deployment
metadata:
  annotations:
    kapp.k14s.io/disable-default-ownership-label-rules: ''
    kapp.k14s.io/disable-default-label-scoping-rules: ''
    kapp.k14s.io/delete-strategy: orphan
    app.github.com/job: '5678'
    app.github.com/ref: refs/tags/v1.2.3
    app.github.com/repo: socialgouv/mano
    app.github.com/run: '1234'
    app.github.com/sha: '0123456'
  labels:
    app: api
    application: mano
    component: mano
    owner: mano
    team: mano
    cert: wildcard
  name: api
  namespace: mano-preprod
spec:
  replicas: 1
  selector:
    matchLabels:
      app: api
  template:
    metadata:
      annotations:
        kapp.k14s.io/disable-default-ownership-label-rules: ''
        kapp.k14s.io/disable-default-label-scoping-rules: ''
        kapp.k14s.io/delete-strategy: orphan
        app.github.com/job: '5678'
        app.github.com/ref: refs/tags/v1.2.3
        app.github.com/repo: socialgouv/mano
        app.github.com/run: '1234'
        app.github.com/sha: '0123456'
      labels:
        app: api
        application: mano
        component: mano
        owner: mano
        team: mano
        cert: wildcard
    spec:
      containers:
        - image: >-
            ghcr.io/socialgouv/mano/api:sha-0123456789abcdefghijklmnopqrstuvwxyz0123
          livenessProbe:
            failureThreshold: 6
            httpGet:
              path: /healthz
              port: http
            initialDelaySeconds: 30
            periodSeconds: 5
            timeoutSeconds: 5
          name: api
          ports:
            - containerPort: 3000
              name: http
          readinessProbe:
            failureThreshold: 15
            httpGet:
              path: /healthz
              port: http
            initialDelaySeconds: 0
            periodSeconds: 5
            successThreshold: 1
            timeoutSeconds: 1
          resources:
            limits:
              cpu: '1'
              memory: 1Gi
            requests:
              cpu: 250m
              memory: 256Mi
          startupProbe:
            failureThreshold: 12
            httpGet:
              path: /healthz
              port: http
            periodSeconds: 5
          volumeMounts:
            - mountPath: /mnt/files
              name: files
          envFrom:
            - secretRef:
                name: azure-pg-user
            - secretRef:
                name: api
          env:
            - name: STORAGE_DIRECTORY
              value: /mnt/files
      imagePullSecrets:
        - name: regcred
      volumes:
        - name: files
          persistentVolumeClaim:
            claimName: files
      initContainers:
        - env:
            - name: WAIT_FOR_RETRIES
              value: '24'
          envFrom:
            - secretRef:
                name: azure-pg-user
          image: ghcr.io/socialgouv/docker/wait-for-postgres:6.56.1
          imagePullPolicy: Always
          name: wait-for-postgres
          resources:
            limits:
              cpu: 20m
              memory: 32Mi
            requests:
              cpu: 5m
              memory: 16Mi
---
apiVersion: bitnami.com/v1alpha1
kind: SealedSecret
metadata:
  annotations:
    sealedsecrets.bitnami.com/cluster-wide: 'true'
    kapp.k14s.io/disable-default-ownership-label-rules: ''
    kapp.k14s.io/disable-default-label-scoping-rules: ''
    kapp.k14s.io/delete-strategy: orphan
    app.github.com/job: '5678'
    app.github.com/ref: refs/tags/v1.2.3
    app.github.com/repo: socialgouv/mano
    app.github.com/run: '1234'
    app.github.com/sha: '0123456'
  name: api
  namespace: mano-preprod
  labels:
    application: mano
    component: mano
    owner: mano
    team: mano
    cert: wildcard
spec:
  encryptedData:
    X_TIPIMAIL_APIUSER: >-
      AgAjJhsQLIyrVrHbKyVpj8oWBqI71zy8zZ/yla93Uzx1ZaY1iNB4olhvzldp8cooons6/8sMI2qmLe5X2sBWtCV7Oc+p7RKGfQmpgi+IDhNv1cxT7PdaGTQXbJzUXBe7Pdkj+EOAKJN4k224I5su8YevT1SDas7l+N97yECWPsrs/LuboMnjjPUuah03HNm6TG4LATJOIQRU7UEPVCqSyMfwb4H6MYjkLU9Awt1bJSjVzgaC80hDknI23Rr7batznFbp77Dt0zNBLPqtaZAAbicA1eGaLnTJNyGZ2tIdWakyv3tmHl+PnlaWEEPoFeKsccdkU8nI+AgwFfkp6jRffxFKDw9Sw5uoMk4XCc3xmE4xybZRtShTf9kSc8qREzS5LFF7bGZkC3SOLd4mbK1+sQ7etuVw2LEZVY79f52fLMBHHGNzXMERZK93WLrmnNJPLRRFpqKyn8QOrqUSMUpKWdlNVanXz6rXiPzhA+cjtIvm50BQX8dEw1OBiFCN9jHgGHM9qd9BeYrVQ8ytqjVJUw5Y0c107vp9kt0K4B5Jy5PXdrWz4+Dd9VzfUCVdsxbWm60pbBY4MwvqTD3mWU3t7vQroZDfQpKx32UBJliHjuUUW6q1cUu+I6MZzfOZXDrcW/MPXb3kiN3eSXRZJ4PRJL1oVwjYGbU1HnQ1OqmCht5IXMh0ft+0wulJGIXBx5YAu3OJNnHiSREP7v/FQNtzV66pZLw5uxIuUpDKfzH44BhgvA==
    X_TIPIMAIL_APIKEY: >-
      AgDNU0wx1SWnpWA9HpF+cGZHyMEilZs67HEDD96ZGKrdZMGUCmfUkv6SRbg1gy+jCFTtfwIi525FImMX9vZQ45bwn0p3/pGLGmLRwvQ2/l26tTsbqrQrw8GWj4biNPS2BVrQlpAw+IJRqKZMW4XbqnN0fhYHN6pJwVEwv+BnY2p9pNGnNtk6ZM9S3cEVk1m0hkK36iytTtZ8UAcSXr8t3humWB12x6oKikvhw6BUyCAJ8qOpguHIflz5M4myPm44yJb5UdQtZNS+LMdItV7PZuBPB8VtDhJ6iPk/bW8dk6gElZrrWBygEtXHfy8kie8Mnytyw9qU2DaDI5ybkAknnyDpBdlOBF+c1mCR4JfYutm70U68Dm7ymR9e6D7BMEsBXW4ikaHGZWGHSsuabhLIP/Iar7WUjx6pQkdtfPn+6Ei7y72M1KxP8GJ79OXtAS4Hjc0AeCcqvs1egTKOHbUOmrBSfAPo3wnLc/bqGyGZ95UlFK5yWF4vENzp8PRve3yT+xa7sOflylfg6rqaYvaCTtZbew8B8YDuHWXDyWP5ZtFP5vZwZ3U8cumxG1GscroTxkt9H3UPEDN6maJKeXZXEzbzCTQ/v6LIOT1ttGv4N1W+64eMoW4ZhkiTdwYXbAEWbYvuLavH0d9scTtOaVym75H75eUNKVrHUv8URnKkHkDnssHKeXjtABUufwSgp35k/J3pfniDakLKFMVNtjUmV1inwOLIQ/d5GSTYVR5XIpxNKA==
    SECRET: >-
      AgDHIAqeo0dIfHXuOEKpxBkqaUz2phtCCIm0NfCI8aowLe1FWQ+sLAAJ1WrS/8jXb8IUZxks5GPEyQ/WBjTneIzT0RATZLJZsoJPYzZ1HlzvL5lHvhPbdvSI1qAT+yZxGzQB7AI3Yflx85ypGxDfWxDT+5oxCAWdfmWRg0zHad0G6BFZkn9oE/i1zmZEuLDhqe2RyLUlmAP6ilDFXjKnO6psz0RrSHUK4axOa80xkRrZOxNxC4zoj0S2+ZiITnv81qqTFCmlnUq3JObLwHhPwQGyyQyRWOfihlHSV9/E6rde9sRfq5Iq+wHPQ0k6ni10u+dMAmDLst/vzx1fILcB5RYPtjWFaDKpUknIZZDCHe/YVDiGDL5RraSWVtvEpwgVMga4KdO5z5EjZowA+iqfvV3sKNokzf2AIpW94Bvq2zXK2ya+yOgABTloI5Jhg/ulKeoLiojqJo6S9CWIq4AirZXgeSxMnQi6PBEdHkoNmrbsu0zvSRxnnOqwoXNEUK2PVwFyUjEdZqwqGayd038iZroFmFxVEebHBVVaoHH/VsVoq915ueM9HHkS+z66/b+iJ1uvg0P7WDLfpb/341GO/8aELBItKuQl7+nXLJh+W18LdZFqIfxbaor1aC3SnkDqf44AJmMyqOT/Gx7nb//0yaMF/a1gWPotNrxDTEs79SCQMlH9rDiYo2WYzNtYAc0WvfvDbedSi224ZwrqG4gPaLbEesUaOuO8yqLHSDM7iRJMNv1lyqo2mGTl
  template:
    metadata:
      annotations:
        sealedsecrets.bitnami.com/cluster-wide: 'true'
        kapp.k14s.io/disable-default-ownership-label-rules: ''
        kapp.k14s.io/disable-default-label-scoping-rules: ''
        kapp.k14s.io/delete-strategy: orphan
        app.github.com/job: '5678'
        app.github.com/ref: refs/tags/v1.2.3
        app.github.com/repo: socialgouv/mano
        app.github.com/run: '1234'
        app.github.com/sha: '0123456'
      name: api
      labels:
        application: mano
        component: mano
        owner: mano
        team: mano
        cert: wildcard
    type: Opaque
---
apiVersion: v1
kind: Service
metadata:
  labels:
    app: api
    application: mano
    component: mano
    owner: mano
    team: mano
    cert: wildcard
  name: api
  annotations:
    kapp.k14s.io/disable-default-ownership-label-rules: ''
    kapp.k14s.io/disable-default-label-scoping-rules: ''
    kapp.k14s.io/delete-strategy: orphan
    app.github.com/job: '5678'
    app.github.com/ref: refs/tags/v1.2.3
    app.github.com/repo: socialgouv/mano
    app.github.com/run: '1234'
    app.github.com/sha: '0123456'
  namespace: mano-preprod
spec:
  ports:
    - name: http
      port: 80
      targetPort: 3000
  selector:
    app: api
  type: ClusterIP
---
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  annotations:
    kubernetes.io/ingress.class: nginx
    kapp.k14s.io/disable-default-ownership-label-rules: ''
    kapp.k14s.io/disable-default-label-scoping-rules: ''
    kapp.k14s.io/delete-strategy: orphan
    app.github.com/job: '5678'
    app.github.com/ref: refs/tags/v1.2.3
    app.github.com/repo: socialgouv/mano
    app.github.com/run: '1234'
    app.github.com/sha: '0123456'
  labels:
    app: api
    application: mano
    component: mano
    owner: mano
    team: mano
    cert: wildcard
  name: api
  namespace: mano-preprod
spec:
  rules:
    - host: mano-preprod.dev2.fabrique.social.gouv.fr
      http:
        paths:
          - backend:
              service:
                name: api
                port:
                  name: http
            path: /
            pathType: Prefix
  tls:
    - hosts:
        - mano-preprod.dev2.fabrique.social.gouv.fr
      secretName: wildcard-crt
---
apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  name: files
  annotations:
    app.github.com/job: '5678'
    app.github.com/ref: refs/tags/v1.2.3
    app.github.com/repo: socialgouv/mano
    app.github.com/run: '1234'
    app.github.com/sha: '0123456'
  labels:
    application: mano
    component: mano
    owner: mano
    team: mano
    cert: wildcard
  namespace: mano-preprod
spec:
  accessModes:
    - ReadWriteMany
  resources:
    requests:
      storage: 20Gi
  selector:
    matchLabels:
      usage: mano-files
  storageClassName: ''
---
apiVersion: v1
kind: PersistentVolume
metadata:
  labels:
    usage: mano-files
    application: mano
    component: mano
    owner: mano
    team: mano
    cert: wildcard
  name: mano-files
  annotations:
    app.github.com/job: '5678'
    app.github.com/ref: refs/tags/v1.2.3
    app.github.com/repo: socialgouv/mano
    app.github.com/run: '1234'
    app.github.com/sha: '0123456'
  namespace: mano-preprod
spec:
  accessModes:
    - ReadWriteMany
  azureFile:
    secretName: azure-mano-volume
    secretNamespace: mano-preprod
    shareName: files
  capacity:
    storage: 20Gi
  persistentVolumeReclaimPolicy: Delete
  storageClassName: ''
---
apiVersion: bitnami.com/v1alpha1
kind: SealedSecret
metadata:
  annotations:
    sealedsecrets.bitnami.com/cluster-wide: 'true'
  name: azure-mano-volume
  namespace: mano-preprod
spec:
  encryptedData:
    azurestorageaccountkey: >-
      AgB2d/mx4rdJz5WJawGd/Dh02k8UjyfcQoyrTaUCH7Kcb9gFPl61BIVsABCtUJDYsFIPmBPNq8u7ihmdD1cvzDbJrbd+fF+83YpGvzHJzrguTiP47ZDT7y3IV1pTP3fXyWnOE4W/cmzNimz7NIiezf3iR3DLavpdG8mw97+66TTeTxnmA9LxWDrIJ8gkna0GnO7TBBlUl/oAyAmG493zfY9DkKot7A1H7efDxqacKxgQDhxG89q/LyWE5ldQp4dBHnD0N6uEsr7/qKtmHvNfSCgJxVQtlYE1nwcgHZ4LHgSHZjHSD5GWqScv/Ft1FmMqomFLFeNUyWbAxGVu4ZNrNrGXOHevPyJKHVGJq8UMfpRZET6ChOUiheIqA6gaHJ7wZ0NnNHq3bydopl3kn/2Q0WnYLE1MrbklJQzXEpNDsu34LlPfmE2+MEzEroPJBa61EDPKT+YsH1M2cceQXha7lqcxWKFtYTPL2JM/DjXd+TFTgL28lGz/frEEIpFnqXR3Y3gjwU4i+JvmeRUGVqLXWVCt4gYwH8H+WeUGF8wjibRdkKc2tm0enA2njoV2mtGvvjr9MEPiL510pJ82o2NYgFIGI/biYUmxde9QYwdaayCHWTdk5ETZZsjqohywxMxjDjZzJ8kS7FTh2y1lKGSJderI8a9AaQrzCmR32ul3v+rZD6nMek7lfZupNQe3+CQkms2OfTwoAub2CNFid81Yn4ula0WaRN/CSFRKhCXwCwTFPco0LkjKRdDGv2KNhEeoHnUAhPS3kMQFSNRh8qmJDWZCaoIAvUPWv4sVnO0rlNuAFuANb3P6Yraz
    azurestorageaccountname: >-
      AgAgS5HrIt+10H9XXlzFDirVh8duoKldmVVEpvOfgDzCmjQawcYDcc0sZByBpZPLzm6SegBm2IKSdFIniq4CQT8lKuMpy0NJGArow7OJffAWkTZfRZRwmIr0bpbPTVnRh3ylXaGDSnLCIUAUPXz3YBwey8UMB0hpNac5UVcw929mTFKuc2xiKwGd7W1KSBLnIUB5KR7wn0MVzk2Vy14JlDDVlk5+vu4e51FhGGzpQLeyV5yBVYijuRR90ApI428qvjJWNGol1QApc/KnvPkbPt9M1gYTXXzfYKB/c+Zok01I1Hdq7CWlvPyh3scY5Uk0/fSwHnzY3LMlsys4Ee7gl9Pn9k+SLz7b3xBfYFUcpkPxSGcN0haBHzO7mp/uAMUZd3PeRu7oExGESdtj4dX4gmz4f7/VhTxwZsjm6xNpMxIYdR+s+lqDIAFOQbvuxzCO8ISkkNtGILuBBS9W2Oh5OUtROVS+xRlyJCxZsUKJqlCCAVd1bxj/LXIcnQNEyfgq0Ic8fvB0tQmZNr3nJsvw4pUTsTRG832cc4FU0M9hKb4k/jTB5kr2TpT14to6QRzQu1u47j+D/1RTDFUmWfyLOU8sCfM5h/uj3eGMgFIkeahNbrpBvArIXUNS16R/LgT6QDBu4A3jo/FTGaTg/ldlfK1SEWeamhfQZJIxkN5fnDp2hu9io9FrNIosSXx6pMXa2tjl9oNpBe79nVDK
  template:
    metadata:
      annotations:
        sealedsecrets.bitnami.com/cluster-wide: 'true'
      name: azure-mano-volume
    type: Opaque
---
apiVersion: apps/v1
kind: Deployment
metadata:
  annotations:
    kapp.k14s.io/disable-default-ownership-label-rules: ''
    kapp.k14s.io/disable-default-label-scoping-rules: ''
    kapp.k14s.io/delete-strategy: orphan
    app.github.com/job: '5678'
    app.github.com/ref: refs/tags/v1.2.3
    app.github.com/repo: socialgouv/mano
    app.github.com/run: '1234'
    app.github.com/sha: '0123456'
  labels:
    app: dashboard
    application: mano
    component: mano
    owner: mano
    team: mano
    cert: wildcard
  name: dashboard
  namespace: mano-preprod
spec:
  replicas: 1
  selector:
    matchLabels:
      app: dashboard
  template:
    metadata:
      annotations:
        kapp.k14s.io/disable-default-ownership-label-rules: ''
        kapp.k14s.io/disable-default-label-scoping-rules: ''
        kapp.k14s.io/delete-strategy: orphan
        app.github.com/job: '5678'
        app.github.com/ref: refs/tags/v1.2.3
        app.github.com/repo: socialgouv/mano
        app.github.com/run: '1234'
        app.github.com/sha: '0123456'
      labels:
        app: dashboard
        application: mano
        component: mano
        owner: mano
        team: mano
        cert: wildcard
    spec:
      containers:
        - image: >-
            ghcr.io/socialgouv/mano/dashboard:sha-0123456789abcdefghijklmnopqrstuvwxyz0123
          livenessProbe:
            failureThreshold: 6
            httpGet:
              path: /healthz
              port: http
            initialDelaySeconds: 30
            periodSeconds: 5
            timeoutSeconds: 5
          name: dashboard
          ports:
            - containerPort: 80
              name: http
          readinessProbe:
            failureThreshold: 15
            httpGet:
              path: /healthz
              port: http
            initialDelaySeconds: 0
            periodSeconds: 5
            successThreshold: 1
            timeoutSeconds: 1
          resources:
            limits:
              cpu: 500m
              memory: 256Mi
            requests:
              cpu: 100m
              memory: 128Mi
          startupProbe:
            failureThreshold: 12
            httpGet:
              path: /healthz
              port: http
            periodSeconds: 5
            initialDelaySeconds: 30
      imagePullSecrets:
        - name: regcred
---
apiVersion: v1
kind: Service
metadata:
  labels:
    app: dashboard
    application: mano
    component: mano
    owner: mano
    team: mano
    cert: wildcard
  name: dashboard
  annotations:
    kapp.k14s.io/disable-default-ownership-label-rules: ''
    kapp.k14s.io/disable-default-label-scoping-rules: ''
    kapp.k14s.io/delete-strategy: orphan
    app.github.com/job: '5678'
    app.github.com/ref: refs/tags/v1.2.3
    app.github.com/repo: socialgouv/mano
    app.github.com/run: '1234'
    app.github.com/sha: '0123456'
  namespace: mano-preprod
spec:
  ports:
    - name: http
      port: 80
      targetPort: 80
  selector:
    app: dashboard
  type: ClusterIP
---
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  annotations:
    kubernetes.io/ingress.class: nginx
    kapp.k14s.io/disable-default-ownership-label-rules: ''
    kapp.k14s.io/disable-default-label-scoping-rules: ''
    kapp.k14s.io/delete-strategy: orphan
    app.github.com/job: '5678'
    app.github.com/ref: refs/tags/v1.2.3
    app.github.com/repo: socialgouv/mano
    app.github.com/run: '1234'
    app.github.com/sha: '0123456'
  labels:
    app: dashboard
    application: mano
    component: mano
    owner: mano
    team: mano
    cert: wildcard
  name: dashboard
  namespace: mano-preprod
spec:
  rules:
    - host: dashboard-mano-preprod.dev2.fabrique.social.gouv.fr
      http:
        paths:
          - backend:
              service:
                name: dashboard
                port:
                  name: http
            path: /
            pathType: Prefix
  tls:
    - hosts:
        - dashboard-mano-preprod.dev2.fabrique.social.gouv.fr
      secretName: wildcard-crt
---
apiVersion: networking.k8s.io/v1
kind: NetworkPolicy
metadata:
  name: netpol-mano-preprod
  namespace: mano-preprod
spec:
  ingress:
    - from:
        - podSelector: {}
    - from:
        - namespaceSelector:
            matchLabels:
              network-policy/source: ingress-controller
    - from:
        - namespaceSelector:
            matchLabels:
              network-policy/source: monitoring
  podSelector: {}
  policyTypes:
    - Ingress
---
apiVersion: apps/v1
kind: Deployment
metadata:
  annotations:
    kapp.k14s.io/disable-default-ownership-label-rules: ''
    kapp.k14s.io/disable-default-label-scoping-rules: ''
    kapp.k14s.io/delete-strategy: orphan
    app.github.com/job: '5678'
    app.github.com/ref: refs/tags/v1.2.3
    app.github.com/repo: socialgouv/mano
    app.github.com/run: '1234'
    app.github.com/sha: '0123456'
  labels:
    app: www
    application: mano
    component: nginx
    owner: mano
    team: mano
    cert: wildcard
  name: www
  namespace: mano-preprod
spec:
  replicas: 1
  selector:
    matchLabels:
      app: www
  template:
    metadata:
      annotations:
        kapp.k14s.io/disable-default-ownership-label-rules: ''
        kapp.k14s.io/disable-default-label-scoping-rules: ''
        kapp.k14s.io/delete-strategy: orphan
        app.github.com/job: '5678'
        app.github.com/ref: refs/tags/v1.2.3
        app.github.com/repo: socialgouv/mano
        app.github.com/run: '1234'
        app.github.com/sha: '0123456'
      labels:
        app: www
        application: mano
        component: nginx
        owner: mano
        team: mano
        cert: wildcard
    spec:
      containers:
        - image: >-
            ghcr.io/socialgouv/mano/website:sha-0123456789abcdefghijklmnopqrstuvwxyz0123
          livenessProbe:
            failureThreshold: 6
            httpGet:
              path: /index.html
              port: http
            initialDelaySeconds: 30
            periodSeconds: 5
            timeoutSeconds: 5
          name: www
          ports:
            - containerPort: 80
              name: http
          readinessProbe:
            failureThreshold: 15
            httpGet:
              path: /index.html
              port: http
            initialDelaySeconds: 0
            periodSeconds: 5
            successThreshold: 1
            timeoutSeconds: 1
          resources:
            limits:
              cpu: 500m
              memory: 128Mi
            requests:
              cpu: 5m
              memory: 32Mi
          startupProbe:
            failureThreshold: 12
            httpGet:
              path: /index.html
              port: http
            periodSeconds: 5
      imagePullSecrets:
        - name: regcred
---
apiVersion: v1
kind: Service
metadata:
  labels:
    app: www
    application: mano
    component: nginx
    owner: mano
    team: mano
    cert: wildcard
  name: www
  annotations:
    kapp.k14s.io/disable-default-ownership-label-rules: ''
    kapp.k14s.io/disable-default-label-scoping-rules: ''
    kapp.k14s.io/delete-strategy: orphan
    app.github.com/job: '5678'
    app.github.com/ref: refs/tags/v1.2.3
    app.github.com/repo: socialgouv/mano
    app.github.com/run: '1234'
    app.github.com/sha: '0123456'
  namespace: mano-preprod
spec:
  ports:
    - name: http
      port: 80
      targetPort: 80
  selector:
    app: www
  type: ClusterIP
---
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  annotations:
    kubernetes.io/ingress.class: nginx
    kapp.k14s.io/disable-default-ownership-label-rules: ''
    kapp.k14s.io/disable-default-label-scoping-rules: ''
    kapp.k14s.io/delete-strategy: orphan
    app.github.com/job: '5678'
    app.github.com/ref: refs/tags/v1.2.3
    app.github.com/repo: socialgouv/mano
    app.github.com/run: '1234'
    app.github.com/sha: '0123456'
  labels:
    app: www
    application: mano
    component: nginx
    owner: mano
    team: mano
    cert: wildcard
  name: www
  namespace: mano-preprod
spec:
  rules:
    - host: app-mano-preprod.dev2.fabrique.social.gouv.fr
      http:
        paths:
          - backend:
              service:
                name: www
                port:
                  name: http
            path: /
            pathType: Prefix
  tls:
    - hosts:
        - app-mano-preprod.dev2.fabrique.social.gouv.fr
      secretName: wildcard-crt
"
`;
